python/cudf/cudf/dataframe/numerical.py

```diff
@@ -246,6 +246,36 @@ def round(self, decimals=0):
             data=Buffer(rounded), mask=mask, dtype=self.dtype
         )
 
+    def clip(self, lower=None, upper=None, inplace=False):
+        mask = None
+        if self.has_null_mask:
+            mask = self.nullmask
+
+        if lower is not None and upper is not None:
+            # check value of lower and upper, and switch them if needed
+            if np.isscalar(lower) and np.isscalar(upper):
+                lower, upper = min(lower, upper), max(lower, upper)
+
+        elif lower is None and upper is None:
+            # if both lower and upper are None, return itself or simply copy it
+            if inplace:
+                return self
+            else:
+                copy = cudautils.copy_array(self.data.mem)
+                return NumericalColumn(
+                    data=Buffer(copy), mask=mask, dtype=self.dtype
+                )
```

Comment:
This is returning the mask by shallow copy which is incorrect behavior here. Should just return `self.copy()` instead here

Reply:
Great point, I think I should also make a copy of null_mask when inplace==False